_episodes/08-putting-it-all-together.md

```diff
@@ -86,209 +89,218 @@ also change the name of the columns in the DataFrame like this:
 
-    plt.show()
+# add an additional ax to the figure
+ax2 = fig.add_axes([0.2, 0.5, 0.4, 0.3]) # inset axes
+ax2.plot(x, y*2, 'r-')
 ~~~
 {: .language-python}
 
-The function `legend()` adds a legend to the figure, and the optional keyword
-arguments change its style. By default [typing just `plt.legend()`], the legend
-is on the upper right corner and has no shadow.
+![Additional axis to line plot](../fig/08_line_plot_inset.png)
 
- The functions `xlabel`, `ylabel`, `title`, `legend`, and many others create text labels. It is good to know that, in addition to the plain text, you may use mathematical notation using a subset of LaTeX language. See [this link](https://matplotlib.org/users/mathtext.html) for more information.
+### Link matplotlib, Pandas and plotnine
 
-
-Like MATLAB, pyplot is stateful; it keeps track of the current figure and
-plotting area, and any plotting functions are directed to those axes. To make
-more than one figure, we use the command `plt.figure()` with an increasing
-figure number inside the parentheses:
+The **Pandas and plotnine packages create matplotlib objects** as well. Hence, using the object based approach provides a consistent workflow and interaction between these packages:
 
 ~~~
-    # This is the first figure
-    plt.figure(1)
-    plt.plot(t, t, 'r--', label='linear')
-
-    plt.legend(loc='upper left', shadow=True, fontsize='x-large')
-    plt.title('This is figure 1')
-
-    plt.show()
-
-    # This is a second figure
-    plt.figure(2)
-    plt.plot(t, t**2, 'bs-', label='square')
+fig, ax1 = plt.subplots() #prepare a matplotlib figure
 
-    plt.legend(loc='upper left', shadow=True, fontsize='x-large')
-    plt.title('This is figure 2')
+surveys.plot("hindfoot_length", "weight", 
+kind="scatter", ax=ax1) # use Pandas for plotting
 
-    plt.show()
+# Provide further adaptations with matplotlib:
+ax1.set_xlabel("Hindfoot length")
+ax1.tick_params(labelsize=16, pad=8)
+fig.suptitle('Scatter plot of weight versus hindfoot length', fontsize=15)
 ~~~
 {: .language-python}
 
-A single figure can also include multiple plots in a grid pattern. The
-`subplot()` command specifies the number of rows, the number of columns, and
-the number of the space in the grid that particular plot is occupying:
+![Extended version of scatter plot surveys](../fig/08_scatter_surveys_extended.png)
 
+To retrieve the matplotlib figure object from plotnine for customization, use the `draw()` function in plotnine:
+
+~~~
+import plotnine as p9
+myplot = (p9.ggplot(data=surveys, 
+mapping=p9.aes(x='hindfoot_length', 
+y='weight')) +
+p9.geom_point())
+
+# convert output plotnine to a matplotlib object
+my_plt_version = myplot.draw()
+
+# Provide further adaptations with matplotlib:
+p9_ax = my_plt_version.axes[0] # each subplot is an item in a list
+p9_ax.set_xlabel("Hindfoot length")
+p9_ax.tick_params(labelsize=16, pad=8)
+p9_ax.set_title('Scatter plot of weight versus hindfoot length', fontsize=15)
+my_plt_version
```

Comment:
`my_plt_version.show()`?

Reply:
Is not required when using the matplotlib inline magic function as far as I know...

Reply:
This works fine on my install of jupyter notebook.